### PR TITLE
Output (n_samples, n_classes) for predict_proba

### PR DESCRIPTION
**Pull request by jwdink**
- Adds `supports_predict_proba` to `Family`
- If the family supports predict_proba, then method is added to GLM
- 'Full' classification is only supported if, additionally, the distribution doesn't have a `total_count` parameter. For distributions like binomial/multinomial, it doesn't make sense to train on / predict classes, because this won't capture heterogeneous counts
- If 'full' classification is supported, you just pass the class-labels and a `LabelEncoder` does the work. if not (e.g. multinomial) you have to pass a `N*num_classes` array for `y` which contains the counts
- For categorical/multinomial, the number of parameters is fewer than the number of classes, since predictions for the final class can be made by leveraging the fact that the sum of the probabilities adds to 1
- Move family aliases from Family to Glm

Resolves https://github.com/strongio/foundry/issues/14

### CODE REVIEW

**Review by andywong36 — CHANGES_REQUESTED** [review]
Some comments below

Adding `('negative_binomial', True)` to L43 of `test_glm.py` means it still works. Which is bizarre - how would a distribution with support over ${0, 1, ..., \infty}$ work?

So I decided to try it, and we have the following:
```
from sklearn.datasets import make_classification
from foundry.glm import Glm
X, y = make_classification(n_informative = 6, n_classes = 4)  # X.shape is (100, 20), y.shape is (100, )
model = Glm("negative_binomial").fit(X, y)
model.predict(X)  # This is (100, 1). Should this have been (100, )? @jwdink in meeting: we should just follow the convention on skl
model.predict_proba(X)  # This is (100, 2). That doesn't make sense to me. 
```

**Comment by jwdink** [comment]
@andywong36 OK, this was tricky, but I have a 2nd stab at this, take a look at the updated description above, and the updated PR.

**Comment by jwdink** [comment]
Now that we have `is_classifier`, I no longer have an excuse not to support R^2 scoring for non-classifiers 😬

**Comment by andywong36** [comment]
@jwdink I'd like to hold off on this review until I complete #16 so I can better understand what's going on.

File: foundry/glm/glm.py [hunk 1]
```diff
@@ -314,10 +409,19 @@ def _get_ydict(self, y: ModelMatrix, sample_weight: Optional[np.ndarray]) -> Dic
                 raise NotImplementedError(
                     f"Unclear how to handle {k}, please report this error to the package maintainer"
                 )
+
+            # special handling for classification:
+            if k == 'value' and self.label_encoder_ is not None:
+                ydict['value'] = self.label_encoder_.transform(ydict['value'])
+                ydict['value'] = to_tensor(ydict['value'], dtype='int', device=_to_kwargs['device'])
```
Comment (andywong36) [inline foundry/glm/glm.py:416]:
```suggestion
                ydict['value'] = to_tensor(ydict['value'], dtype=torch.long, device=_to_kwargs['device'])
```

I was getting an error. Unsure what this does, this modification seemed to let the code run.

**Review by andywong36 — CHANGES_REQUESTED** [review]
